Add method getworstaudio #233

Divyansh2512 opened the pull request:
Returns the lowest bitrate audio stream object

Divyansh2512 committed:
Add method getworstaudio #233
Returns the lowest bitrate audio stream object

ritiek commented on pafy/backend_shared.py:
```diff
@@ -401,6 +401,26 @@ def _sortkey(x, keybitrate=0, keyftype=0):
+            keybitrate = int(x.rawbitrate)
+            keyftype = preftype == x.extension
+            strict, nonstrict = (keyftype, keybitrate), (keybitrate, keyftype)
+            return strict if ftypestrict else nonstrict
```
This nested function is exactly the same as defined in `getbestaudio` here:
https://github.com/mps-youtube/pafy/blob/933d08fc63d4ba1d82b7f10b3f8730ab9a664168/pafy/backend_shared.py#L384-L402

Perhaps we should move this sort key function to the current class scope instead of nesting the exact same definition inside two different methods. We could then use this sort key method from all other dependent methods without having to define it twice. This will help with code re-usability!

ritiek reviewed (commented):
Thanks for the PR!

I've left a review. We could merge this in once it has been addressed.

Divyansh2512 committed:
Add a new method `sortaudiokey` for reusability.
This method returns a tuple and is used in two methods `getbestaudio`
and `getworstaudio` ensuring its reusability.

ritiek reviewed (approved):
LGTM. Thank you once again! We should also add other methods such as `getworstvideo`, etc.  Perhaps in another PR? :)
